packages/smooth_app/lib/pages/personalized_ranking_page.dart

```diff
@@ -61,21 +61,34 @@ class _PersonalizedRankingPageState extends State<PersonalizedRankingPage>
     context.watch<LocalDatabase>();
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
 
-    Future<void> _addToList() async {
+    Future<void> addToLists() async {
       final LocalDatabase localDatabase = context.read<LocalDatabase>();
       final DaoProductList daoProductList = DaoProductList(localDatabase);
       await ProductListUserDialogHelper(daoProductList)
           .showBulkInsertUserListsDialog(
         context,
         widget.barcodes,
       );
+      if (!mounted) {
```

Comment:
Here you're not even sure you added products. What if the user clicked on "cancel"? What if the added products were already in the list(s)?